common/components/schemas.py

```
@@ -21,8 +23,16 @@ def get_nonan_index(value: list) -> int | None:
     -----
     This function iterates over the list to find the first value that is not NaN.
     """
+
+    def isna(value):
+        if not value:
+            return True
+        if isinstance(value, float) and isnan(value):
+            return True
+        return False
```

Ruff

Return the condition `bool(isinstance(value, float) and isnan(value))` directly

Replace with `return bool(isinstance(value, float) and isnan(value))`

(SIM103)